src/analysis/asymptotics/negligible.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import analysis.asymptotics.asymptotics
+import analysis.asymptotics.specific_asymptotics
+import data.polynomial.eval
+
+/-!
+# Negligible Functions
+
+This file defines a predicate `negligible f` on functions `f` from `ℕ` to a normed field `𝕜`.
+
+The main theorem is `negligible_polynomial_mul` that says the product of a polynomial
+  and a negligible function is still a negligible function
+-/
+
+namespace asymptotics
+
+/-- Definition of negligible functions over an arbitrary `normed_field`.
+  Note that the second function always has type `ℕ → ℝ`, which generally gives better lemmas. -/
+def negligible {𝕜 : Type*} [normed_ring 𝕜] (f : ℕ → 𝕜) :=
+∀ (c : ℤ), is_O f (λ n, (n : ℝ) ^ c) filter.at_top
+
+section normed_ring
+
+variables {R : Type*} [normed_ring R]
+variables {f g : ℕ → R}
+
+lemma negligible_of_is_O (hg : negligible g)
+  (h : is_O f g filter.at_top) : negligible f :=
```

Comment:
If you write it as
```suggestion
lemma is_O.trans_negligible (h : is_O f g filter.at_top) (hg : negligible g) :
  negligible f :=
```
then users will be able to write `h.trans_negligible hg`. And I hope that some day we'll fix `calc` to accept things like that as a `@[trans]` lemmas.
You can also add
```lean
alias is_O.trans_negligible ← asymptotics.negligible.is_O_mono
```
to have dot notation on the other argument as well.